packages/near-membrane-base/src/membrane.ts

```diff
@@ -341,20 +344,108 @@ export function createMembraneMarshall(
     }
 
     function proxyMaskFunction<T extends Function>(func: Function, maskFunc: T): T {
+        let nearMembraneSymbolFlag = false;
+        let lastProxyTrapCalled = ProxyHandlerTraps.None;
         const proxy = new ProxyCtor(maskFunc, {
             apply(_target: T, thisArg: any, args: any[]) {
+                lastProxyTrapCalled = ProxyHandlerTraps.Apply;
                 if (thisArg === proxy || thisArg === maskFunc) {
                     thisArg = func;
                 }
                 return ReflectApply(func, thisArg, args);
             },
             construct(_target: T, args: any[], newTarget: Function) {
+                lastProxyTrapCalled = ProxyHandlerTraps.Construct;
                 if (newTarget === proxy || newTarget === maskFunc) {
                     newTarget = func;
                 }
                 return ReflectConstruct(func, args, newTarget);
             },
-        });
+            defineProperty(target: ProxyTarget, key: PropertyKey, desc: PropertyDescriptor) {
+                lastProxyTrapCalled = ProxyHandlerTraps.DefineProperty;
+                if (key === LOCKER_NEAR_MEMBRANE_PROXY_MASKED_SYMBOL) {
+                    throw new TypeErrorCtor(ERR_ILLEGAL_PROPERTY_ACCESS);
+                }
+                return ReflectDefineProperty(target, key, desc);
+            },
+            deleteProperty(target: ProxyTarget, key: PropertyKey) {
+                lastProxyTrapCalled = ProxyHandlerTraps.GetOwnPropertyDescriptor;
+                return ReflectDeleteProperty(target, key);
+            },
+            get(target: ProxyTarget, key: PropertyKey, receiver: any) {
+                // Only allow accessing near-membrane symbol values if the
+                // BoundaryProxyHandler.has trap has been called immediately before
+                // and the symbol does not exist.
+                nearMembraneSymbolFlag &&= lastProxyTrapCalled === ProxyHandlerTraps.Has;
+                lastProxyTrapCalled = ProxyHandlerTraps.Get;
+                const isProxyMaskedSymbol = key === LOCKER_NEAR_MEMBRANE_PROXY_MASKED_SYMBOL;
+                if (nearMembraneSymbolFlag) {
+                    // Exit without performing a [[Get]] for near-membrane symbols
+                    // because we know when the nearMembraneSymbolFlag is ON that
+                    // there is no shadowed symbol value.
+                    if (isProxyMaskedSymbol) {
+                        return true;
+                    }
+                }
+                const result = ReflectGet(target, key, receiver);
+                if (result !== undefined && isProxyMaskedSymbol) {
+                    throw new TypeErrorCtor(ERR_ILLEGAL_PROPERTY_ACCESS);
```

Comment:
why throwing instead of just returning `undefined`?

Reply:
Because this is a guard for when a user is faking this. Instead of silently failing when we detect this kind of level of forgery since it is for the identification of our masking and proxies we throw a hard error. No user code should ever be doing this (trying to pass off this symbol as their own).

Reply:
Essentially before we were playing nice and when, in the case of forging we would simply return `undefined` and not recognize it, but identification of masked and proxied values is becoming more critical (so we don't want to fail the check). Now there is no wiggle room for possible forgeries and if a user is trying to forge something we should error and say "No, don't do that".